elastic/beats
# Add dependencies-report script

- Bug

## What does this PR do?

Adds a script to report dependencies in a csv file by running `dev-tools/dependencies-report -csv $OUTPUT_FILE`

Sample output:

```
name,url,version,revision,license
4d63.com/tz,https://4d63.com/tz,v1.1.1,6d37baae851b,MIT
cloud.google.com/go,https://cloud.google.com/go,v0.51.0,,Apache-2.0
cloud.google.com/go/pubsub,https://cloud.google.com/go/pubsub,v1.0.1,,Apache-2.0
cloud.google.com/go/storage,https://cloud.google.com/go/storage,v1.0.0,,Apache-2.0
code.cloudfoundry.org/go-loggregator,https://code.cloudfoundry.org/go-loggregator,v7.4.0,,Apache-2.0
github.com/Azure/azure-event-hubs-go/v3,https://github.com/Azure/azure-event-hubs-go,v3.1.2,,MIT
github.com/Azure/azure-sdk-for-go,https://github.com/Azure/azure-sdk-for-go,v37.1.0,,Apache-2.0
github.com/Azure/azure-storage-blob-go,https://github.com/Azure/azure-storage-blob-go,v0.8.0,,MIT
github.com/Azure/go-autorest/autorest,https://github.com/Azure/go-autorest,v0.9.4,,Apache-2.0
```


## Why is it important?

With the removal of vendor we don't create a dependency report anymore, which is required as part of the unified release process.

## Checklist



~~- [ ] My code follows the style guidelines of this project~~
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## How to test this PR locally

run `$ dev-tools/dependencies-report -csv test.csv`. On success check `test.csv`

## Related issues


-
